### ISSUE TITLE
False positive color-no-invalid-hex

### ISSUE THREAD

**Issue by crudo** [stylelint/stylelint#2305]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Bug

> Which rule, if any, is this issue related to?

`color-no-invalid-hex`

> What CSS is needed to reproduce this issue?

```css
body { color: #fff#; }
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "color-no-invalid-hex": true
  }
}
```

> Which version of stylelint are you using?

`7.7.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint --config .stylelintrc bug.css`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No.

> What did you expect to happen?

Unexpected invalid hex color error.

> What actually happened (e.g. what warnings or errors you are getting)?

None.

**Comment by hudochenkov** [comment]
Thank you for reporting!

I believe this rule works as intended. It check color as part of value in different properties. It doesn't check value for property `color`. It searches for hex colors (`#` + letters and/or digits) in properties values and checks them. Later `#` isn't a part of color. In your case there is a valid color, but invalid value for property `color`. It's not a subject to check for `color-no-invalid-hex`.

This won't trigger a warning either:

```css
a {
    color: green #f00;
}
```

**Comment by crudo** [comment]
Hi @hudochenkov, yes, you are right. I was not sure if it belongs to this rule, but it seems to be some king of a bug anyway. Because the css is not valid.

**Comment by hudochenkov** [comment]
@jeddy3:
> It is beyond the scope of stylelint's core package to effectively validate against the CSS spec. Please investigate [csstree](https://github.com/csstree/csstree) and [css-values](https://github.com/ben-eb/css-values) for this functionality. csstree already has a [stylelint plugin](https://github.com/csstree/stylelint-validator), and css-values needs one to be developed.

**Comment by selcet** [comment]
Unfortunately, no handle for such cases:
```css
 body { color: fff; }
```
or
```css
 body { color: ffffff; }
```
